core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -121,46 +123,36 @@ class LoadBalancerService(config: WhiskConfig, entityStore: EntityStore)(implici
      *
      * @param msg is the kafka message payload as Json
      */
-    private def processCompletion(aid: ActivationId, tid: TransactionId, response: Option[WhiskActivation], forced: Boolean = false) = {
-        logging.info(this, s"received active ack for '$aid' (forced = $forced)")(tid)
+    private def processCompletion(aid: ActivationId, tid: TransactionId, response: Option[WhiskActivation]): Unit = {
         activationById.remove(aid) match {
             case Some(entry @ ActivationEntry(_, subject, invokerIndex, _, p)) =>
+                logging.info(this, s"received active ack for '$aid' (forced = ${response.isEmpty})")(tid)
                 activationByInvoker.getOrElseUpdate(invokerIndex, new TrieSet[ActivationEntry]).remove(entry)
                 activationBySubject.getOrElseUpdate(subject, new TrieSet[ActivationEntry]).remove(entry)
-                response.foreach(p.trySuccess(_))
-                logging.info(this, s"processed active response for '$aid'")(tid)
+                p.trySuccess(response.map(Right(_)).getOrElse(Left(aid)))
```

Comment:
There is: `opt.toRight({{leftValue}}`, so in this case: `p.trySuccess(response.toRight(aid))`